spec/requests/server_render_check_spec.rb

```diff
@@ -0,0 +1,17 @@
+require "rails_helper"
+
+describe "Server Rendering" do
+  it "generates server rendered HTML if server rendering enabled" do
+    get root_path
+    html_nodes = Nokogiri::HTML(response.body)
+    expect(html_nodes.css("div#App-react-component-0").children.size).to eq(1)
+    expect(html_nodes.css("div#App-react-component-0 h2").text)
+      .to include("Comments")
+  end
+
+  it "generates no server rendered HTML if server rendering not enabled" do
```

Comment:
@josiasds should we have fragment caching and server rendering enabled for the simple example? Let's have a link on the simple page to render with no server rendering. We can have the prerender be a query param for the get and if false, then no pre-rendering.